src/traces/barpolar/hover.js

```diff
@@ -0,0 +1,78 @@
+/**
+* Copyright 2012-2018, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var barHover = require('../bar/hover');
+var makeHoverPointText = require('../scatterpolar/hover').makeHoverPointText;
+var Color = require('../../components/color')
+var Fx = require('../../components/fx');
+var Lib = require('../../lib');
+var isPtInsidePolygon = require('../../plots/polar/helpers').isPtInsidePolygon;
+
+module.exports = function hoverPoints(pointData, xval, yval, hovermode) {
+    var cd = pointData.cd;
+    var trace = cd[0].trace;
+    var t = cd[0].t;
+
+    var subplot = pointData.subplot;
+    var radialAxis = subplot.radialAxis;
+    var angularAxis = subplot.angularAxis;
+    var xa = subplot.xaxis;
+    var ya = subplot.yaxis;
+    var inboxFn = subplot.vangles ? isPtInsidePolygon : Lib.isPtInsideSector;
+
+    if(radialAxis.range[0] > radialAxis.range[1]) {
+        xval = -xval;
+        yval = -yval;
+    }
+
+    var rVal = Math.abs(radialAxis.g2p(Math.sqrt(xval * xval + yval * yval)));
+    var thetaVal = Math.atan2(yval, xval);
+
+    // TODO add padding around sector to show labels,
+    // when hovering "close to" them
+    var distFn = function(di) {
+        var rBnds = [di.s0, di.s1].map(radialAxis.c2p);
+        var thetaBnds = [di.p0, di.p1].map(angularAxis.c2g).map(Lib.rad2deg);
```

Comment:
🐎 is there a reason not to do the inverse of these `map`s on `rVal` and `thetaVal`? Then in here I guess all you'd need to do is restrict `thetaBnds` to the same range?

Relatedly, what if your bars wrap around more than 360º, there must be a cut in `thetaVal` somewhere, what if there's a bar that crosses the cut, will its "in bounds" appear to be the entire rest of the circle?

Reply:
Ah I see - yes, looks like `isAngleInsideSector` accounts for this aliasing, so long as none of the conversions you do to `thetaBnds` do any wrapping of their own 🎉 but this requires periodicity of 360, hence this mapping is required to be on this side. Well OK, I guess we won't have enough data here to worry *that* much about 🐎 (so this is all nonblocking) - but if we do want to improve this it would probably just require modifying `inboxFn` to work with the calcdata periodicity rather than 360.

Reply:
Ha sorry, I missed that you *did* clean this up in https://github.com/plotly/plotly.js/pull/2954/commits/7ae3b7cd11211e52467a4a70218fb3c859b70356, and then you removed the last `map` inside `distFn` in https://github.com/plotly/plotly.js/pull/2954/commits/e93897473bec9cf27f7e4a63cf85ce55ac4de1b6 - beautiful.